inference/core/workflows/core_steps/transformations/detections_merge/v1.py

```diff
@@ -0,0 +1,131 @@
+from typing import Any, Dict, List, Literal, Optional, Type
+from uuid import uuid4
+
+import numpy as np
+import supervision as sv
+from pydantic import ConfigDict, Field
+
+from inference.core.workflows.execution_engine.entities.base import OutputDefinition
+from inference.core.workflows.execution_engine.entities.types import (
+    INSTANCE_SEGMENTATION_PREDICTION_KIND,
+    KEYPOINT_DETECTION_PREDICTION_KIND,
+    OBJECT_DETECTION_PREDICTION_KIND,
+    Selector,
+)
+from inference.core.workflows.prototypes.block import (
+    BlockResult,
+    WorkflowBlock,
+    WorkflowBlockManifest,
+)
+
+OUTPUT_KEY: str = "predictions"
+
+SHORT_DESCRIPTION = "Merge multiple detections into a single bounding box."
+LONG_DESCRIPTION = """
+The `DetectionsMerge` block combines multiple detections into a single bounding box that encompasses all input detections.
+This is useful when you want to:
+- Merge overlapping or nearby detections of the same object
+- Create a single region that contains multiple detected objects
+- Simplify multiple detections into one larger detection
+
+The resulting detection will have:
+- A bounding box that contains all input detections
+- The class_id and confidence from the first detection in the input
+"""
+
+
+class DetectionsMergeManifest(WorkflowBlockManifest):
+    model_config = ConfigDict(
+        json_schema_extra={
+            "name": "Detections Merge",
+            "version": "v1",
+            "short_description": SHORT_DESCRIPTION,
+            "long_description": LONG_DESCRIPTION,
+            "license": "Apache-2.0",
+            "block_type": "transformation",
+            "ui_manifest": {
+                "section": "transformation",
+                "icon": "fal fa-object-union",
+                "blockPriority": 5,
+            },
+        }
+    )
+    type: Literal["roboflow_core/detections_merge@v1"]
+    predictions: Selector(
+        kind=[
+            OBJECT_DETECTION_PREDICTION_KIND,
```

Comment:
not sure if this should be only 
`OBJECT_DETECTION_PREDICTION_KIND` and `INSTANCE_SEGMENTATION_PREDICTION_KIND` (tested with those two), but I think KEYPOINT_DETECTION_PREDICTION_KIND should also be OK / we have bounding box info via detections?